README.md

```diff
@@ -25,3 +25,14 @@ The app deployed as part of this solution does the following:
 
 The agent crashes whenever it encounters an unrecoverable error with the metadata APIs.
 This agent is not production hardened yet and so use it with caution.
+
+## Graceful terminations for regular pods (Non-system pods)
+
+The pods that are not in the kube-system are called **regular pods** in this agent.
+By default, regular pods are deleted immediately before deleting system pods.
+If you want to delete regular pods gracefully, please add `--system-pod-grace-period=n` in arguments according to the following rules:
+
+- If targeted VM is Preemptible VM, specify `n` with a value from `0s` to `15s`.
```

Comment:
For Preemptible, the timeout of the regular pod is set to a value of less than 30 seconds (slightly less, NOT equals to 30 seconds). This is because the process of determining `NodeTerminationState.TerminationTime` (1) and the process of determining the timeout of the regular pod (2) are executed at different times.

1. Add 30 seconds to the time the `maintenanceEvent` or `preemptedEvent` was subscribed from Metadata.
https://github.com/GoogleCloudPlatform/k8s-node-termination-handler/blob/d56ae034967302518f8613d7333de5610965fd23/termination/gceTerminationHandler.go#L116

2. Subtract the time `processNodeState` was called from `TerminationTime`.
https://github.com/GoogleCloudPlatform/k8s-node-termination-handler/blob/d56ae034967302518f8613d7333de5610965fd23/termination/handler.go#L58

Therefore, specify `15s` for `--system-pod-grace-period` is NOT work. Delete regular pods gracefully, you probably need to specify `n` with a value from `0s` to `14s`. .... The current rules are very complex. For the sake of clarity, I think this README is sound good.

Reply:
Thanks for pointing this out. That makes sense. I just fixed it and mentioned why it does not go well. Could you take a look at that again?

Reply:
Thanks! +1

Reply:
Just to clarify the main reason that the value needs to be between 0s and 14s for Preemptible VM is due to this [check](https://github.com/GoogleCloudPlatform/k8s-node-termination-handler/blob/master/termination/eviction.go#L78-L80).

I think the time between the GCE maintenance event being triggered and `EvictPods()` being called should be minimal.